iredis/client.py

```diff
@@ -37,29 +36,12 @@ class Client:
     """
 
     def __init__(self, host, port, db, password=None):
-        self.host = host
-        self.port = port
-        self.db = db
-        if config.decode:
-            self.connection = Connection(
-                host=self.host,
-                port=self.port,
-                db=self.db,
-                password=password,
-                encoding=config.decode,
-                decode_responses=True,
-                encoding_errors="replace",
-                socket_keepalive=config.socket_keepalive,
-            )
-        else:
-            self.connection = Connection(
-                host=self.host,
-                port=self.port,
-                db=self.db,
-                password=password,
-                decode_responses=False,
-                socket_keepalive=config.socket_keepalive,
-            )
+        self.host = config.connection_kwargs.get("host")
+        self.port = config.connection_kwargs.get("port")
+        self.db = config.connection_kwargs.get("db")
+        self.path = config.connection_kwargs.get("path")
+        self.connection = config.connection_class(**config.connection_kwargs)
+
```

Comment:
The decode-responses arg is missed here, I prefer pass arts explicitly here

Reply:
The decode-responses arg is handled in gather_connection_kwargs